src/releaseProvider.ts

```diff
@@ -0,0 +1,77 @@
+import { env, ExtensionContext } from "vscode";
+import * as vscode from "vscode";
+import * as semver from "semver";
+
+const versionKey = "metals-server-version";
+
+export class ReleaseProvider {
+  private readonly state: typeof this.context.globalState;
+
+  constructor(
+    private readonly context: ExtensionContext,
+    // TODO switch to extension version maybe?
+    private readonly currentServerVersion: string
+  ) {
+    this.state = context.globalState;
+  }
+
+  showWhatsNewIfNeeded() {
+    if (this.shouldShowPanel) {
+      this.showPanel();
+    }
+  }
+
+  private showPanel() {
+    const previousValueOpt: string | undefined = this.state.get(versionKey);
+
+    const previousValue = previousValueOpt || this.currentServerVersion;
```

Comment:
So in case we didn't save the version (mechanism was never invoked) we would show the panel for the user? If that is the case I think it's a good idea.

Reply:
Currently if there was no version we didn't show anything. But showing current release notes makes sense, I'll change that

Reply:
update: if there was no version before, show release notes for current version